src/app/components/problems/problem-set/problem-set.component.ts

```diff
@@ -110,6 +130,14 @@ export class ProblemSetComponent implements OnInit {
             this.questions = paginatedQuestions.results;
             this.questionsSource = new MatTableDataSource(this.questions);
         });
+        this.importExportService.downloadAllQuestions().subscribe(questions => {
```

Comment:
It should download the questions only once when the user clicks on the download button.
It's really inefficient to do this and will reduce the performance of the website.